src/ceph-disk

```diff
@@ -2191,11 +2353,65 @@ def mount_activate(
 
     path = mount(dev=dev, fstype=fstype, options=mount_options)
 
+    # check if the disk is deactive, change the journal owner, group
+    # mode for correct user and group.
+    if os.path.exists(os.path.join(path, 'deactive')):
+        # logging to syslog will help us easy to know udev triggered failure
+        if not reactivate:
+            unmount(path)
+            syslog.syslog(syslog.LOG_ERR, 'OSD deactivated! reactivate with: --reactivate')
+            raise Error('OSD deactivated! reactivate with: --reactivate')
+        # flag to activate a deactive osd.
+        deactive = True
+    else:
+        deactive = False
+
+    journal_dev = os.path.realpath(os.path.join(path,'journal'))
+    try:
+        if get_ceph_user() == 'ceph':
+            command(
+                [
+                'chown', '-R', 'ceph:ceph',
+                journal_dev,
+                ],
+            )
+            command(
+                [
+                'chmod', '660',
+                journal_dev,
+                ]
+            )
+        if dmcrypt:
+            journal_dev_dmcrypt = os.path.realpath(os.path.join(path,'journal_dmcrypt'))
+            command(
+                [
+                'chown', '-R', 'ceph:ceph',
+                journal_dev_dmcrypt,
+                ],
+            )
+            command(
+                [
+                'chmod', '660',
+                journal_dev_dmcrypt,
+                ]
+            )
+    except OSError:
+        pass
+
     osd_id = None
     cluster = None
     try:
         (osd_id, cluster) = activate(path, activate_key_template, init)
 
+        # Now active successfully
+        # If we got reactivate and deactive, remove the deactive file
+        if deactive and reactivate:
+            try:
```

Comment:
I mean what if you replace
<pre>
+        if deactive and reactivate:
+            try:
+                os.remove(os.path.join(path, 'deactive'))
+                LOG.info('Remove `deactive` file.')
+            except OSError:
+                raise Error('Cannot remove `deactive` file!')
</pre>
with just
<pre>
+                os.remove(os.path.join(path, 'deactive'))
+                LOG.info('Remove `deactive` file.')
</pre>
If the file removal fails, it will throw an exception. There does not seem to be a need to catch the exception and throw another one. I'm also concerned that the raise Error() may hide the real cause of the error.